identity-server/templates/src/TemplateWebApp/HostingExtensions.cs

```diff
@@ -0,0 +1,112 @@
+
+            ?? throw new InvalidOperationException("IdentityProvider configuration is missing or invalid.");
+
+        JwtSecurityTokenHandler.DefaultMapInboundClaims = false;
+
+        // The __Host- prefix ensures the cookie is host-only, requires Secure and Path=/ attributes
+        var hostCookiePrefix = "__Host-";
+
+        builder.Services.AddAuthentication(options =>
+        {
+            options.DefaultScheme = CookieAuthenticationDefaults.AuthenticationScheme;
+            options.DefaultChallengeScheme = OpenIdConnectDefaults.AuthenticationScheme;
+        })
+            .AddCookie(options =>
+            {
+                options.Cookie.Name = hostCookiePrefix + "TemplateWebApp";
+                options.Cookie.Path = "/";
+                options.Cookie.HttpOnly = true;
+                options.Cookie.SecurePolicy = CookieSecurePolicy.Always;
+                options.Cookie.SameSite = SameSiteMode.Lax;
+                options.ExpireTimeSpan = TimeSpan.FromHours(8);
+                options.SlidingExpiration = true;
+            })
+            .AddOpenIdConnect(OpenIdConnectDefaults.AuthenticationScheme, options =>
+            {
+                options.Events.OnRemoteFailure = context =>
+                {
+                    context.Response.Redirect("/Error?message=" + context.Failure?.Message ?? "A remote failure occured");
+                    context.HandleResponse();
+                    return Task.CompletedTask;
+                };
+
+                options.Authority = identityProviderConfig.Authority;
+                options.ClientId = identityProviderConfig.ClientId;
+                options.ClientSecret = identityProviderConfig.ClientSecret;
+
+                options.ResponseType = OpenIdConnectResponseType.Code;
+                options.UsePkce = true;
+
+                options.Scope.Clear();
+                options.Scope.Add("offline_access");
+
+                // Add additional scopes from the configuration
+                identityProviderConfig.Scopes.Split(' ', StringSplitOptions.RemoveEmptyEntries)
+                    .ToList().ForEach(scope => options.Scope.Add(scope));
+
+                options.GetClaimsFromUserInfoEndpoint = true;
+                options.SaveTokens = true;
+
+                options.TokenValidationParameters = new TokenValidationParameters
+                {
+                    NameClaimType = JwtClaimTypes.Name,
+                    RoleClaimType = JwtClaimTypes.Role,
+                };
+
+                options.DisableTelemetry = true;
+            });
+
+        builder.Services.AddOpenIdConnectAccessTokenManagement();
+
+        // In production, use a secure location for storing keys, such as Azure Key Vault or AWS Secrets Manager.
+        // Make sure to secure the keys at rest.
+        builder.Services.AddDataProtection()
+            .SetApplicationName("TemplateWebApp")
+            .PersistKeysToFileSystem(new DirectoryInfo("/keys"));
```

Comment:
Running on my machine, this threw an exception that /keys was a readonly filesystem. "./keys" worked. Maybe this is platform specific (I'm mac, you're PC, right?)